students/alex_whitty/lesson4/trigrams.py

```diff
@@ -0,0 +1,83 @@
+import random
+import string
+#!/usr/bin/env python3
+def parse_file():
+    '''
+    Takes a hardcoded file name and returns a list of words
+    '''
+    # Setup a translator to remove punctuation
+    translator = str.maketrans('', '', string.punctuation)
+    # Setup an empty list to contain the full list of words in the text
+    full_list = []
+    with open('double_take.txt', 'r') as f:
+        for line in f:
+            # This will take each line, remove punctuation and then split it
+            # into a list of words and attach this list to the full_list
+            # note the use of "extend()" instead of "append()"
+            # TODO: Figure out how to remove carriage returns
+            full_list.extend(line.translate(translator).split(' '))
+    return full_list
+
+
+
+
+trigrams = {("When", "the"): ['Travelers'],
+            ("the", "Travelers"): ["from"],
+            ("Travelers", "from"): ["Outer"],
+            ("from", "Outer"): ["space"],
+            ("Outer", "Space"): ["dug"],
+            ("Space", "dug"): ["into"],
+            ("dug", "into"): ["the"],
+            ("into", "the"): ["pile"],
+            }
+
+
+
+def build_trigrams(words):
+
+    trigrams = {}
+
+    for i in range(len(words) - 2):
+
+
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+
+        if pair not in trigrams:
+
+            trigrams[pair] = [follower]
+        else:
+            trigrams[pair].append(follower)
+
+
+    return trigrams
+
+# build up the dict here!
+
+def build_dict(words):
+    starter = random.choice(list(words.keys("from", "Outer")))
+    print(starter)
+    # Start the text with the random key
+    # now turned into a string
+    new_text = ' '.join(starter)
+
+    # Now, pick a random word from the list
+    # associated to the key
+    key_value = words["space"]
+    next_word = random.choice(key_value)
+
+    # Add next_word to new_text
+    new_text += 'Outer' + next_word
+    print(new_text)
+
+    # Form the next round, using the second
+    # word of the starter and next_word
+    # and repeat the process
+
+
+if __name__ == "__main__":
+    words = "When the Travelers from outer Space dug into the pile".split()
```

Comment:
You should update this one to call _parse_file()_ to produce your list of words.